types/src/channels/embed.ts

```diff
@@ -0,0 +1,66 @@
+// deno-lint-ignore-file camelcase no-empty-interface
+
+export enum EmbedType {
+  RICH = "rich",
+  IMAGE = "image",
+  VIDEO = "video",
+  GIFV = "gifv",
+  ARTICLE = "article",
+  LINK = "link",
+}
+
+export interface EmbedVideoPayload {
+  url?: string;
+  proxy_url?: string;
+  height?: number;
+  width?: number;
+}
+
+export interface EmbedImagePayload {
+  url?: string;
+  proxy_url?: string;
+  height?: number;
+  width?: number;
+}
+
+export interface EmbedThumbnailPayload extends EmbedImagePayload {}
```

Comment:
wouldn't `export type EmbedThumbnailPayload = EmbedImagePayload` work here?

Reply:
yeah it would work but i'm not sure whether discord will add some other properties to the thumbnail payload

Reply:
We could always change it later. Idk if the linter warns about this. If it does it's prob better to make it a type alias

Reply:
hmm yeah linter does warn about it, i'll change those to type alias